presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryConfig.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.bigquery;
+
+import com.facebook.airlift.configuration.Config;
+import com.facebook.airlift.configuration.ConfigDescription;
+import com.google.auth.oauth2.GoogleCredentials;
+
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.Min;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+public class BigQueryConfig
+{
+    public static final int DEFAULT_MAX_READ_ROWS_RETRIES = 3;
+
+    private Optional<String> credentialsKey = Optional.empty();
+    private Optional<String> credentialsFile = Optional.empty();
+    private Optional<String> projectId = Optional.empty();
+    private Optional<String> parentProjectId = Optional.empty();
+    private OptionalInt parallelism = OptionalInt.empty();
+    private boolean viewsEnabled;
+    private Optional<String> viewMaterializationProject = Optional.empty();
+    private Optional<String> viewMaterializationDataset = Optional.empty();
+    private int maxReadRowsRetries = DEFAULT_MAX_READ_ROWS_RETRIES;
+
+    @AssertTrue(message = "Exactly one of 'bigquery.credentials-key' or 'bigquery.credentials-file' must be specified, or the default GoogleCredentials could be created")
+    public boolean isCredentialsConfigurationValid()
```

Comment:
it's registered as a custom validation method so that at initialization time to boostrap bigquery connector, configuration factory calls`validateRegisteredConfigurationProvider` method to validate the credentials